Bump sphinx rtd theme to version 1.0

tobiaskaestner opened the pull request.

rexut commented:
You should sign all your commits with GPG --> https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

`git commit -S ...` <-- the upercase S

or permanently --> `git config --global commit.gpgsign true`

rexut commented on requirements.txt:
```diff
@@ -1 +1 @@
-sphinx_rtd_theme~=0.5
\ No newline at end of file
+sphinx_rtd_theme~=1.0
```
As long as Zephyr still stay on any version lower than 1.0 we will wait to merge and also we need to spend some more time to qualify our theme against all the changes in the new RTD theme.

rexut reviewed (changes requested):
we'l wait until Zephyr v3.x in Bridle

rexut committed:
Remove key-value dependency of theme_docsets.items

In preperation to support more external docset entries
with the same context as decoded by the list key the
configuration value config[1] now have to provide the
document root that may have an equal prefix, ex.:

before:
  {{"name": bridle,
    "config": {"Bridle", "index"}},
   {"name": bridle-api,
    "config": {"Bridle API", "doxygen/html/index"}}}
becomes:
  <a href=".../bridle/index.html">Bridle</a>
  <a href=".../bridle-api/doxygen/html/index.html">Bridle API</a>

  ... but that's incorrect and have to be as ...

now:
  {{"name": bridle,
    "config": {"Bridle", "bridle/index"}},
   {"name": bridle-api,
    "config": {"Bridle API", "bridle/doxygen/html/index"}}}
becomes:
  <a href=".../bridle/index.html">Bridle</a>
  <a href=".../bridle/doxygen/html/index.html">Bridle API</a>

Signed-off-by: Stephan Linz <linz@li-pro.net>

rexut commented:
manual merged in via cherry-pick from c85b23024d9d6264c01ba7d48c986e95bbda1bd4 -- thx
